modules/gitrepo/tree_path.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package gitrepo
+
+import (
+	"context"
+
+	"code.gitea.io/gitea/modules/git"
+)
+
+func GetTreePathLatestCommit(ctx context.Context, repo Repository, commitID, treePath string) (*git.Commit, error) {
```

Comment:
Why in `gitrepo` package but not `git` package?

Do you have a clear plan or roadmap for these two packages? Or just keep making them more messy like other legacy/problematic code?

Reply:
My plan is that `modules/git` will be the low level of `modules/gitrepo`. `modules/gitrepo` is only for managed repositories but `modules/git` will be a generate git wrap library which can be used for managed repositories or temporary repositories. And in `modules/gitrepo`, the repository can be reused by context.

Reply:
Then you shouldn't put "git command calls" in "gitrepo", right?

Reply:
16506ad8f5eab8da87ea36ef9a95b150d675d16d

Reply:
If you are not able to remember the plan&design, please please please write documents and comments, in case you would forget how to write the right code next time.

Reply:
Why do you think `gitrepo` should not invoke `git command calls`? Since `git command calls` are all the basics of almost all git operations, that will be unavoidable.

Reply:
> `modules/gitrepo` is only for managed repositories but `modules/git` will be a general git wrap library which can be used for managed repositories or temporary repositories.

It is answered by your description. Then the basic git operations should be provided by the "general git library", but not by a context-repo-manager.